### PR TITLE
Init spring boot app registration on back

### PR DESCRIPTION
**Pull request by VChorney**

### CODE REVIEW

File: src/main/java/com/quizter/config/WebConfig.java [hunk 1]
```diff
@@ -12,5 +17,7 @@
     @Override
     public void addViewControllers(final ViewControllerRegistry registry) {
         registry.addViewController("/").setViewName("home-page");
+        registry.addViewController("/login").setViewName("login-page");
+//        registry.addViewController("/resetPassword").setViewName("reset-password-page");
```
Comment (itevas) [inline src/main/java/com/quizter/config/WebConfig.java:21]: delete commented line

**Review by itevas — APPROVED** [review]

File: src/main/java/com/quizter/controller/RegistrationController.java [hunk 2]
```diff
@@ -1,27 +1,36 @@
...
+    public String registrationForm(Model model) {
+        model.addAttribute("user", new RegistrationUserDto());
+        return "registration-page";
     }
 
     @PostMapping
-    public ResponseEntity<MessageResponse> registerUser(@RequestBody RegistrationUserDto registrationDto) {
-        userService.registerUser(registrationDto);
-        return ResponseEntity.ok(new MessageResponse("You've  been registered"));
+    public String registration(@ModelAttribute RegistrationUserDto user, @PathVariable("role") Role role) {
```
Comment (relyativus) [inline src/main/java/com/quizter/controller/RegistrationController.java:32]: this endpoint should be restful.

File: src/main/java/com/quizter/controller/ResetPasswordController.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    MailWebService mailWebService;
+
+    @GetMapping("/resetPassword")
+    public String resetPasswordPage() {
+        return "reset-password-page";
+    }
+
+    @PostMapping("/resetPassword")
+    public String resetPassword(@RequestParam String username, Model model) {
+        Optional<User> userOptional = userService.findUserByEmail(username);
```
Comment (relyativus) [inline src/main/java/com/quizter/controller/ResetPasswordController.java:37]: this endpoint should be restful

File: src/main/java/com/quizter/entity/PasswordResetToken.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    Long id;
+
+    String token;
+
+    @OneToOne(targetEntity = User.class, fetch = FetchType.EAGER)
+    @JoinColumn(nullable = false, name = "user_id")
+    User user;
+
+    Date expiryDate;
```
Comment (relyativus) [inline src/main/java/com/quizter/entity/PasswordResetToken.java:33]: Don't use java.util.Date. Use java.time.Instant. Date is legacy, most methods in it are deprecated

File: src/main/java/com/quizter/repository/PasswordRepository.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.quizter.repository;
+
+import com.quizter.entity.PasswordResetToken;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface PasswordRepository extends CrudRepository<PasswordResetToken, Long> {
+
+    PasswordResetToken save(PasswordResetToken passwordResetToken);
```
Comment (relyativus) [inline src/main/java/com/quizter/repository/PasswordRepository.java:10]: Looks like you override standard save method from CrudRepository

File: src/main/java/com/quizter/service/mailsender/MailSender.java [hunk 6]
```diff
@@ -9,4 +11,6 @@
 
     void sendMessageWithTemplate(String recipient, String subject, Map<String, Object> model);
 
+//    void sendPasswordResetToken(User user);
```
Comment (relyativus) [inline src/main/java/com/quizter/service/mailsender/MailSender.java:14]: remove commented code

File: src/main/java/com/quizter/service/mailsender/implementation/MailSenderImpl.java [hunk 7]
```diff
@@ -56,4 +59,13 @@ public void sendMessageWithTemplate(String recipient, String subject, Map<String
         javaMailSender.send(message);
     }
 
+//    @Override
```
Comment (relyativus) [inline src/main/java/com/quizter/service/mailsender/implementation/MailSenderImpl.java:62]: remove commented code

**Review by relyativus — CHANGES_REQUESTED** [review]

File: src/main/java/com/quizter/controller/ResetPasswordController.java (hunk 3 above)
Comment (itevas) [reply src/main/java/com/quizter/controller/ResetPasswordController.java:37]: It's mine, I'm working on it in forgotPassword branch

File: src/main/java/com/quizter/entity/PasswordResetToken.java (hunk 4 above)
Comment (itevas) [reply src/main/java/com/quizter/entity/PasswordResetToken.java:33]: It's mine either, I'll fix it in my branch

File: src/main/java/com/quizter/repository/PasswordRepository.java (hunk 5 above)
Comment (itevas) [reply src/main/java/com/quizter/repository/PasswordRepository.java:10]: It's mine again, Ok)

**Pull request closed without merging**